webpack/assets/javascripts/react_app/components/ForemanModal/ForemanModal.js

```diff
@@ -6,15 +6,30 @@ import ForemanModalHeader from './subcomponents/ForemanModalHeader';
 import { extractModalNodes } from './helpers';
 
 const ForemanModal = props => {
-  const { id, title, onClose, isOpen, children, ...propsToPassDown } = props;
+  const {
+    id,
+    title,
+    onClose,
+    isOpen,
+    children,
+    onSubmitStart,
+    onSubmitStop,
+    isSubmitting,
+    submitProps,
+    ...propsToPassDown
+  } = props;
   // Extract header and footer from children, if provided
   const { headerChild, footerChild, otherChildren } = extractModalNodes(
     children
   );
   const context = {
     isOpen,
     onClose,
+    isSubmitting,
+    onSubmitStart,
+    onSubmitStop,
     title,
+    submitProps,
   };
 
   let headerToRender = null; // if no headerChild and no title prop, then no <ForemanModalHeader>
```

Comment:
If `submitProps` are supplied then maybe it should render a `<ForemanModal.Footer />` for you, as it does here with the Header.  Wouldn't want to swallow submitProps because I forgot to put in the `<ForemanModal.Footer />` child.